src/components/dapp-staking/statistics/Withdraw.vue

```diff
@@ -0,0 +1,165 @@
+    <div class="tw-flex tw-flex-col tw-items-center">
+      <FormatBalance :balance="totalToWithdraw" class="tw-flex tw-text-2xl tw-font-bold" />
+    </div>
+    <div class="tw-flex tw-flex-row tw-items-baseline tw-float-right">
+      <div class="tw-cursor-pointer tw-mr-4" @click="showModal = true">
+        {{ $t('dappStaking.chunks') }}
+      </div>
+      <Button v-if="canWithdraw" :primary="false" class="tw-mt-4" @click="withdraw()">
+        {{ $t('dappStaking.withdraw') }}
+      </Button>
+    </div>
+
+    <ChunksModal
+      v-if="showModal"
+      v-model:isOpen="showModal"
+      :unlocking-chunks="unlockingChunks"
+      :max-unlocking-chunks="maxUnlockingChunks"
+    />
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent, onUnmounted, computed, ref, watch } from 'vue';
+import BN from 'bn.js';
+import { useApi } from 'src/hooks';
+import { useStore } from 'src/store';
+import { u32 } from '@polkadot/types';
+import { VoidFn } from '@polkadot/api/types';
+import { Balance, EraIndex } from '@polkadot/types/interfaces';
+import { Codec } from '@polkadot/types/types';
+import Button from 'src/components/common/Button.vue';
+import { WithdrawParameters } from 'src/store/dapp-staking/actions';
+import FormatBalance from 'components/balance/FormatBalance.vue';
+import ChunksModal from './ChunksModal.vue';
+import { useUnbondWithdraw } from 'src/hooks/useUnbondWithdraw';
+
+export default defineComponent({
+  components: {
+    Button,
+    FormatBalance,
+    ChunksModal,
+  },
+  setup() {
+    const { api } = useApi();
+    const store = useStore();
+    const selectedAccountAddress = computed(() => store.getters['general/selectedAccountAddress']);
+    const unlockingChunksCount = computed(() => store.getters['dapps/getUnlockingChunks']);
+    const maxUnlockingChunks = computed(() => store.getters['dapps/getMaxUnlockingChunks']);
+    const unlockingChunks = ref<ChunkInfo[]>();
+    const canWithdraw = ref<boolean>(false);
+    const totalToWithdraw = ref<BN>(new BN(0));
+    const showModal = ref<boolean>(false);
+    const { canUnbondWithdraw } = useUnbondWithdraw();
+
+    const withdraw = async (): Promise<void> => {
+      const result = await store.dispatch('dapps/withdrawUnbonded', {
+        api: api?.value,
+        senderAddress: selectedAccountAddress.value,
+      } as WithdrawParameters);
+    };
+
+    const subscribeToEraChange = async (): Promise<VoidFn | undefined> => {
+      const unsub = (await api?.value?.query.dappsStaking.currentEra(async (era: u32) => {
+        console.log('new era', era.toNumber(), selectedAccountAddress.value);
+        await getChunks(era);
+      })) as VoidFn | undefined;
+
+      return unsub;
+    };
+
+    const unsub = subscribeToEraChange();
+
+    const getChunks = async (era: u32) => {
+      if (!canUnbondWithdraw) {
+        return;
+      }
+
+      const unbondingInfo =
+        await api?.value?.query.dappsStaking.unbondingInfoStorage<UnbondingInfo>(
```

Comment:
![Screen Shot 2021-12-14 at 1 55 15 PM](https://user-images.githubusercontent.com/40356749/146002545-042b4df2-3155-4462-b487-ede9c448ade1.png)

The `unbondingInfoStorage` property is undefined on networks that does not yet implement the unbounding period feature on Astar network.

```
Uncaught (in promise) TypeError: _b.query.dappsStaking.unbondingInfoStorage is not a function
    at eval (webpack-internal:///./node_modules/@quasar/app/lib/webpack/loader.js.transform-quasar-imports.js!./node_modules/ts-loader/index.js??clonedRuleSet-3.use[0]!./node_modules/@quasar/app/lib/webpack/loader.vue.auto-import-quasar.js??ruleSet[0].use[0]!./node_modules/vue-loader/dist/index.js??ruleSet[0].use[1]!./src/components/dapp-staking/statistics/Withdraw.vue?vue&type=script&lang=ts:69)
    at Generator.next (<anonymous>)
    at eval (webpack-internal:///./node_modules/@quasar/app/lib/webpack/loader.js.transform-quasar-imports.js!./node_modules/ts-loader/index.js??clonedRuleSet-3.use[0]!./node_modules/@quasar/app/lib/webpack/loader.vue.auto-import-quasar.js??ruleSet[0].use[0]!./node_modules/vue-loader/dist/index.js??ruleSet[0].use[1]!./src/components/dapp-staking/statistics/Withdraw.vue?vue&type=script&lang=ts:21)
    at new Promise (<anonymous>)
    at __awaiter (webpack-internal:///./node_modules/@quasar/app/lib/webpack/loader.js.transform-quasar-imports.js!./node_modules/ts-loader/index.js??clonedRuleSet-3.use[0]!./node_modules/@quasar/app/lib/webpack/loader.vue.auto-import-quasar.js??ruleSet[0].use[0]!./node_modules/vue-loader/dist/index.js??ruleSet[0].use[1]!./src/components/dapp-staking/statistics/Withdraw.vue?vue&type=script&lang=ts:17)
    at getChunks (webpack-internal:///./node_modules/@quasar/app/lib/webpack/loader.js.transform-quasar-imports.js!./node_modules/ts-loader/index.js??clonedRuleSet-3.use[0]!./node_modules/@quasar/app/lib/webpack/loader.vue.auto-import-quasar.js??ruleSet[0].use[0]!./node_modules/vue-loader/dist/index.js??ruleSet[0].use[1]!./src/components/dapp-staking/statistics/Withdraw.vue?vue&type=script&lang=ts:64)
    at eval (webpack-internal:///./node_modules/@quasar/app/lib/webpack/loader.js.transform-quasar-imports.js!./node_modules/ts-loader/index.js??clonedRuleSet-3.use[0]!./node_modules/@quasar/app/lib/webpack/loader.vue.auto-import-quasar.js??ruleSet[0].use[0]!./node_modules/vue-loader/dist/index.js??ruleSet[0].use[1]!./src/components/dapp-staking/statistics/Withdraw.vue?vue&type=script&lang=ts:59)
    at Generator.next (<anonymous>)
    at eval (webpack-internal:///./node_modules/@quasar/app/lib/webpack/loader.js.transform-quasar-imports.js!./node_modules/ts-loader/index.js??clonedRuleSet-3.use[0]!./node_modules/@quasar/app/lib/webpack/loader.vue.auto-import-quasar.js??ruleSet[0].use[0]!./node_modules/vue-loader/dist/index.js??ruleSet[0].use[1]!./src/components/dapp-staking/statistics/Withdraw.vue?vue&type=script&lang=ts:21)
    at new Promise (<anonymous>)
```

Reply:
Nice catch. I improved a logic for checking if unbonding is supported a little bit. Previously logic was checking a network name and now checks network feature to determine if unbonding is supported.